Tribler/Test/Core/CreditMining/test_creditmining_sys.py

```diff
@@ -0,0 +1,376 @@
+        self.bsettings.initial_logging_interval = 3
+
+    def setUpPreSession(self):
+        super(TestBoostingManagerSys, self).setUpPreSession()
+
+        self.config.set_torrent_checking(True)
+        self.config.set_megacache(True)
+        self.config.set_dispersy(True)
+        self.config.set_mainline_dht(True)
+        self.config.set_torrent_store(True)
+        self.config.set_enable_torrent_search(True)
+        self.config.set_enable_channel_search(True)
+        # self.config.set_torrent_collecting(True)
+        self.config.set_libtorrent(True)
+        self.config.set_dht_torrent_collecting(True)
+        # self.config.set_enable_metadata(True)
+        # self.config.set_http_api_enabled(True)
+        # self.config.set_upgrader_enabled(False)
+
+    def tearDown(self):
+        self.boosting_manager.shutdown()
+        self.boosting_manager.del_instance()
+        os.remove('None')
+
+        super(TestBoostingManagerSys, self).tearDown()
+
+class TestBoostingManagerSysRSS(TestBoostingManagerSys):
+    """
+    testing class for RSS (dummy) source
+    """
+    def __init__(self, *argv, **kwargs):
+        super(TestBoostingManagerSysRSS, self).__init__(*argv, **kwargs)
+
+    def setUp(self, autoload_discovery=True):
+        super(TestBoostingManagerSysRSS, self).setUp()
+
+        # Prepare test_rss.xml file, replace the port with a random one
+        self.file_server_port = get_random_port()
+        files_path = os.path.join(self.session_base_dir, 'http_torrent_files')
+        os.mkdir(files_path)
+
+        with open(os.path.join(TESTS_DATA_DIR, 'test_rss_cm.xml'), 'r') as source_xml,\
+                open(os.path.join(files_path, 'test_rss_cm.xml'), 'w') as destination_xml:
+            for line in source_xml:
+                destination_xml.write(line.replace('RANDOMPORT', str(self.file_server_port)))
+
+        shutil.copyfile(TORRENT_FILE, os.path.join(files_path, 'ubuntu.torrent'))
+
+        exist_torrent_url = "http://localhost:RANDOMPORT/existbuntu.torrent".replace(
+            "RANDOMPORT", str(self.file_server_port))
+        exist_infohash = sha1(exist_torrent_url).digest()
+
+        shutil.copyfile(TORRENT_FILE, os.path.join(self.boosting_manager.settings.credit_mining_path,
+                                      'rss-%s.torrent' % exist_infohash.encode('hex')))
+        self.setUpFileServer(self.file_server_port, files_path)
+
+        # now the rss should be at :
+        # http://localhost:port/test_rss_cm.xml
+        # which resides in sessiondir/http_torrent_files
+
+    def test_rss(self):
+        """
+        test rss source
+        """
+        url = 'http://localhost:%s/test_rss_cm.xml' %self.file_server_port
+        self.boosting_manager.add_source(url)
+
+        rss_obj = self.boosting_manager.get_source_object(url)
+        count = 0
+        while not rss_obj.ready and count < 10:
+            rss_obj = self.boosting_manager.get_source_object(url)
+            time.sleep(5)
```

Comment:
We should not add more `time.sleep` statements but instead, use deferred objects for these tasks. Could you investigate how we can use deferreds here?